c7n/resources/cw.py

```diff
@@ -114,6 +119,135 @@ def get_dimensions(self, resource):
         return [{'Name': 'RuleName', 'Value': resource['Name']}]
 
 
+@EventRule.filter_registry.register('event-rule-target')
+class EventRuleTargetFilter(ChildResourceFilter):
+    """
+    Filter event rules by their targets
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-event-rules-with-no-targets
+              resource: aws.event-rule
+              filters:
+                - type: event-rule-target
+                  key: Arn
+                  value: absent
+    """
+
+    RelatedResource = "c7n.resources.cw.EventRuleTarget"
+    RelatedIdsExpression = 'Name'
+    AnnotationKey = "EventRuleTargets"
+
+    schema = type_schema('event-rule-target', rinherit=ValueFilter.schema)
+    permissions = ('events:ListTargetsByRule',)
+
+
+@EventRule.filter_registry.register('invalid-targets')
+class ValidEventRuleTargetFilter(ChildResourceFilter):
+    """
+    Filter event rules for invalid targets
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-event-rules-with-invalid-targets
+              resource: aws.event-rule
+              filters:
+                - type: invalid-targets
```

Comment:
probably need an option to only have the filter match if `all` the targets are invalid (aka totally safe to delete), where as in the case of partial it would still be useful for notification purposes.

Reply:
agreed 👍